src/main/java/com/amazonaws/encryptionsdk/kmsv2/AwsKmsMrkAwareMasterKeyProvider.java

```diff
@@ -44,31 +41,29 @@
 
   private final boolean isDiscovery_;
   private final DiscoveryFilter discoveryFilter_;
-  private final String discoveryMrkRegion_;
+  private final Region discoveryMrkRegion_;
 
-  private final KmsMasterKeyProvider.RegionalClientSupplier regionalClientSupplier_;
-  private final String defaultRegion_;
+  private final RegionalClientSupplier regionalClientSupplier_;
+  private final Region defaultRegion_;
 
   public static class Builder implements Cloneable {
-    private String defaultRegion_ = getSdkDefaultRegion();
-    private Optional<KmsMasterKeyProvider.RegionalClientSupplier> regionalClientSupplier_ =
-        Optional.empty();
-    private AWSKMSClientBuilder templateBuilder_ = null;
+    private Region defaultRegion_ = getSdkDefaultRegion();
+
+    private Supplier<KmsClientBuilder> builderSupplier_ = null;
+    private RegionalClientSupplier regionalClientSupplier_ = null;
+
     private DiscoveryFilter discoveryFilter_ = null;
-    private String discoveryMrkRegion_ = this.defaultRegion_;
+    private Region discoveryMrkRegion_ = this.defaultRegion_;
 
     Builder() {
       // Default access: Don't allow outside classes to extend this class
     }
 
     public Builder clone() {
```

Comment:
This isn't blocking, but this is a good opportunity to move from Cloneable to a copy constructor pattern, i.e. implement a toBuilder() instead of this clone. The Cloneable pattern is confusing and easy to get wrong, and I believe the SDK already has customers familiar with the toBuilder() pattern.

Reply:
That's a great idea, especially since the SDKv2 clients no longer support the `Cloneable` pattern either. I will probably push this change to a later CR though since its more of a functional change.